core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -71,6 +78,57 @@ public static CloseableIterable<FileScanTask> splitFiles(
     return CloseableIterable.combine(splitTasks, tasks);
   }
 
+  public static <T extends FileScanTask> Iterable<CombinedScanTask> planTasks(
```

Comment:
If we accept a list, what about returning a list as well? We access tasks by index in `SparkBatch`.

Reply:
Done.